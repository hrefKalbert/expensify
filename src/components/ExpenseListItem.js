import React from 'react'
import { Link } from 'react-router-dom'

// const ExpenseListItems = (props) => 
//   <ol>
//     {props.expenses.map(cur => (
//       <div key={cur.id}>
//         <li><h3>{cur.description}</h3></li>
//         <li>Amount: {cur.amount}</li>
//         <li>Created at: {cur.createdAt}</li>
//       </div>
//       ) )}
//   </ol>

//   const mapStateToProps = (state) => ({
//     expenses: state.expenses,
//     filters: state.filters
//   }) 

//   export default connect(mapStateToProps)(ExpenseListItems)

const ExpenseListItems = ({ description, amount, createdAt, id }) =>
  <div>
    <Link to={`/edit/${id}`}>
      <h1>{description}</h1>
    </Link>
    <p>Amount: {amount}</p>
    <p>Created at: {createdAt}</p>

  </div>

export default ExpenseListItems