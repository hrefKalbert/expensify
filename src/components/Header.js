import React from 'react'
import {NavLink} from 'react-router-dom'


const Header = () =>
  <header>
    <h1>Expensify</h1>
    <NavLink activeClassName="is-active" to="/" exact={true}>Dash board</NavLink>
    <NavLink activeClassName="is-active" to="/create">Go to Create page</NavLink>
  </header>
 

export default Header