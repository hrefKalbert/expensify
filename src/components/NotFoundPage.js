import React from 'react'
import {Link} from 'react-router-dom'

const NotFound = () =>
  <div>
    <h3>401! NOT FOUND! <Link to="/">Home</Link></h3>
  </div>

  export default NotFound