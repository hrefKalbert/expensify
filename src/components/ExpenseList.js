import React from 'react'
import ExpenseListItem from './ExpenseListItem'
import { connect } from 'react-redux'
import {getVisibleExpenses} from '../selectors/expenses'

export const ExpenseList = (props) => 
  <div>
    {
      props.expenses.length === 0 ? 
        <p>No items in the list</p> :
      props.expenses.map(cur => 
          <ExpenseListItem 
          key={cur.id}
          {...cur}
           />
           )

    }
  </div>

  const mapStateToProps = (state) => ({
    expenses: getVisibleExpenses(state.expenses, state.filters)
  
  })



  export default connect(mapStateToProps)(ExpenseList)
  