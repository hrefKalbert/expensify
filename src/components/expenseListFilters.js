import React from 'react'
import { connect } from 'react-redux'
import ExpenseList from './ExpenseList';
import 'react-dates/initialize'
// import 'react-dates/lib/css/_datepicker.css' Moved this one to the app js file because of testing errors
import { DateRangePicker } from 'react-dates'
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate)
    this.props.setEndDate(endDate)
  }

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({
      calendarFocused
    }))
  }

  onSortChange = (e) => {

    e.target.value === 'date' ? this.props.sortByDate() : this.props.sortByAmount()

  }

  onTextChange = (e) => this.props.setTextFilter(e.target.value)
  render() {
    return (
      <div>
        <input type="text" value={this.props.filters.text} onChange={this.onTextChange} />
        <select name="" id="" value={this.props.filters.sortBy} onChange={this.onSortChange}>
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker 
          startDate={this.props.filters.startDate}
          startDateId="uniqeStartDateId"
          endDate={this.props.filters.endDate}
          endDateId="uniqueEndDateId"
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={()=> false}
          showClearDates={true}
        />
      </div>
    )
  }
}

const mapDispatchToProps = () => ({
  setTextFilter: text => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: ()=> dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate))
})


const mapStateToProps = (state) => ({
  filters: state.filters
})
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)