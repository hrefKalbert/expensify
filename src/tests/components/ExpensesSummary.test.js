import React from 'react'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import {shallow } from 'enzyme'

test('should correctly render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235}/>)
  expect(wrapper).toMatchSnapshot();
})

test('should correctly render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={23545353}/>)
  expect(wrapper).toMatchSnapshot();
})