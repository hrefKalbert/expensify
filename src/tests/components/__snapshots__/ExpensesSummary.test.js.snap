// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `<div />`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `<div />`;
